wp-parsely.php

```diff
@@ -47,12 +226,43 @@ function maybe_load_plugin() {
 	 *
 	 * add_filter( 'wpvip_parsely_load_mu', '__return_false' );
 	 */
-	if ( ! apply_filters( 'wpvip_parsely_load_mu', get_option( '_wpvip_parsely_mu' ) === '1' ) ) {
+
+	// The activation of this plugin has been queued in this request.
+	$self_managed_plugin_is_queued_for_activation = is_queued_for_activation();
+
+	// Check if wp-parsely has already been loaded, for example when the user
+	// has a self-managed wp-parsely installation.
+	$self_managed_plugin_has_loaded = in_array( PARSELY_PLUGIN_SIGNATURE, get_option( 'active_plugins', [] ) );
```

Comment:
This isn't a safe enough protection to use to abort early if the plugin is already loaded. The plugin signature does not have to be `wp-parsely/wp-parsely.php`, it could be anything like `wp-parsely-2/wp-parsely.php` for example.

We need to return to the safety of `class_exists( 'Parsely' ) || class_exists( 'Parsely\Parsely' )` type checks.

Reply:
In my testing, `class_exists( 'Parsely' ) || class_exists( 'Parsely\Parsely' )` never actually worked when the user was using a self-managed plugin. I believe it is because this code is executed before the plugin was loaded.

Reply:
Please see the changes in my latest commit which address all of the concerns that you brought up. I have tested the scenarios and it seems to be all just fine. 

The `class_exists( 'Parsely' ) || class_exists( 'Parsely\Parsely' )` logic does work as expected when a user is doing a direct require in `client-mu-plugins/plugin-loader.php` or putting their wp-parsely directly in `client-mu-plugins`.

If this class is not found, it reverse to the methodology of checking the `active_plugins` list, but this time it uses a regex pattern to identify anything that looks like it's a wp-parsely plugin. e.g. `wp-parsely2/wp-parsely.php`, `wp-parsely-1.10/wp-parsely.php`, etc...

I believe that if there is any more of a wild Parse.ly plugin out there, then this would be such a small statistical anomaly that it's not really all that relevant for this scope.

Here is the commit log:

* Added back in `class_exists( 'Parsely' ) || class_exists( 'Parsely\Parsely' )` logic to account for when the plugin is loaded through `client-mu-plugins`.
  * Added function: `has_plugin_signature` which uses a regex pattern to determine if a plugin is loaded that looks like wp-parsely.
  * Changed constant `PARSELY_PLUGIN_SIGNATURE` to `PARSELY_PLUGIN_SIGNATURE_REGEX` to represent the pattern which a wp-parsely plugin could look like.
  * Updated tests and added tests for `has_plugin_signature`
  * Thoroughly tested all scenarios so that I know it works.